content/scala-cli.md

```diff
@@ -0,0 +1,150 @@
+
+
+### Specification
+
+ In order to be able to expand the functionality of Scala CLI and yet use the same core to power the `scala` command, we propose to include both `scala` and `scala-cli` commands in the installation package. Scala CLI already has a feature to limit accessible sub-commands based the binary name (all sub-commands in `scala-cli`, and a curated list in `scala`). Later, we can include more and more features from `scala-cli` into `scala`. 
+
+The sub-commands necessary to include in the `scala` to match the functionalities of current commands:
+ - `compile`
+ - `run`
+ - `repl`
+ - `doc`
+
+On top of that, we think that the following user-facing sub-commands should also be included:
+ - `clean` - to rebuild project from start without any cached steps
+ - `setup-ide` - to control the setup of BSP for IDE integration
+ - `doctor` - to analyze if everything is installed properly
+
+We also suggest to include additional sub-commands by default: 
+ - `fmt` - to format the code using scalafmt
+ - `test` - to run tests included in the code
+ - `package` - to package the code into various package formats: JAR, "fat"  be, executable  be or even native application or docker image
+ - `shebang` - a command useful for scripting, designed to be included in the "shebang" section of scripts
+ - `export` - transform current project to sbt / Mill project using the same settings as provided. Useful to evolve prototypes into bigger projects
+
+Each of these commands expands what the current scripts offer and can be discussed separately. We can even open a dedicated SIP for each of them.
+
+Beyond that, `scala-cli` offers multiple sub-commands needed to manage itself (e.g. `update`) or its components (e.g. the Bloop server). In most cases,
+these are not user-facing, but still handy. We can elaborate in more detail on what those commands are and why we need them, if necessary.
+
+Scala CLI can also be configured with ["using directives"](https://scala-cli.virtuslab.org/docs/guides/using-directives) - a comment-based configuration syntax that should be placed at the top of Scala files. This allows for self-containing examples within one file since most of the configuration can be provided either from the command line or via using directives (command line has precedence). This is a game changer for use cases like scripting, reproduction, or within the academic scope.
+
+We have described the motivation, syntax and implementation basis in the [dedicated pre-SIP](https://contributors.scala-lang.org/t/pre-sip-using-directives/5700). Currently, we recommend to write using directives as comments, so making them part of the language specification is not necessary at this stage. Moreover, the new `scala` command could ignore using directives in the initial version, however we strongly suggest to include comment-based using directives from the start. 
+
+### Compatibility
```

Comment:
I believe another incompatibility that should be mentioned here for the sake of completeness is that it may not be possible to have distinct Scala CLI scripts in the same directory, whereas this is possible with the current `scala` command line tool.

If I understand correctly, Scala CLI loads all the .scala files of the current directory, which means that it is impossible to have, in the same directory, one script with `using scala 3.1.3` and another script with `using scala 2.13.8`. They would have to live in separate directories.

Reply:
fwiw, I haven't observed that to be the case with my `scala-cli shebang` based scripts.

Reply:
> If I understand correctly, Scala CLI loads all the .scala files of the current directory

This is the case only if `.` is provided as one of the inputs. When user runs `scala-cli A.scala B.scala` the effect is the same regardless what sits in the current directory.

To sum up: it is possible to have `A.scala` using Scala `2.13` and `B.scala` using Scala `3.1` in the same directory without any conflicts as long as both are not run together (using e.g. `scala-cli .`).

Reply:
Great, thank you for clarifying, I’m marking the discussion as resolved :)